LazyPRM/ LazyPRM creates 4294967284 (close to max unsigned int) states and ignores termination condition

I am using LazyPRM to plan paths with an octomap as collision object in the planning scene. Most of the time it works great! and I get good results within the configured planning time. Sometimes, the planner greatly exceeds the max planning time and creates in one query 4294967284 new states. Is this maybe related to #852?

Thanks in advance for your help!

My planner config is:
```
LazyPRM
      type: geometric::LazyPRM
      multi_query_planning_enabled: true
      store_planner_data: false
      load_planner_data: false
```

And a sample of my output looks like this (the last query causes the descriped weird behavior):
```
[ INFO] [1663926051.523048331]: Planner configuration 'arm[LazyPRM]' will use planner 'geometric::LazyPRM'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1663926051.544336791]: arm/arm[LazyPRM]: Starting planning with 6537 states already in datastructure
[ INFO] [1663926051.559041942]: arm/arm[LazyPRM]: Created 1 states
[ INFO] [1663926051.559090062]: Solution found in 0.035874 seconds
[ INFO] [1663926051.565494578]: SimpleSetup: Path simplification took 0.002998 seconds and changed from 3 to 2 states
[ INFO] [1663926051.617874978]: Fake execution of trajectory
[ INFO] [1663926053.218046857]: Completed trajectory execution with status SUCCEEDED ...
[ INFO] [1663926053.223446775]: Planner configuration 'arm[LazyPRM]' will use planner 'geometric::LazyPRM'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1663926053.244450562]: arm/arm[LazyPRM]: Starting planning with 6540 states already in datastructure
[ INFO] [1663926053.246350873]: arm/arm[LazyPRM]: Created 1 states
[ INFO] [1663926053.246419178]: Solution found in 0.022804 seconds
[ INFO] [1663926053.250311363]: SimpleSetup: Path simplification took 0.000001 seconds and changed from 2 to 2 states
[ INFO] [1663926053.318304362]: Fake execution of trajectory
[ INFO] [1663926054.817978762]: Completed trajectory execution with status SUCCEEDED ...
[ INFO] [1663926054.823426521]: Planner configuration 'arm[LazyPRM]' will use planner 'geometric::LazyPRM'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1663926054.895198352]: arm/arm[LazyPRM]: Starting planning with 6543 states already in datastructure
[ INFO] [1663926054.926337994]: arm/arm[LazyPRM]: Created 39 states
[ INFO] [1663926054.926391144]: No solution found after 0.102832 seconds
[ INFO] [1663926054.934644008]: Unable to solve the planning problem
[ERROR] [1663926054.934768465]: Could not compute plan successfully
[ERROR] [1663926054.935534345]: Error code: PLANNING_FAILED
[ INFO] [1663926054.939960326]: Planner configuration 'arm[LazyPRM]' will use planner 'geometric::LazyPRM'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1663926055.021393441]: arm/arm[LazyPRM]: Starting planning with 6584 states already in datastructure
[ INFO] [1663926055.040251215]: arm/arm[LazyPRM]: Created 89 states
[ INFO] [1663926055.040306949]: No solution found after 0.100206 seconds
[ INFO] [1663926055.061319605]: Unable to solve the planning problem
[ERROR] [1663926055.061405021]: Could not compute plan successfully
[ERROR] [1663926055.061973040]: Error code: PLANNING_FAILED
[ INFO] [1663926055.066205980]: Planner configuration 'arm[LazyPRM]' will use planner 'geometric::LazyPRM'. Additional configuration parameters will be set when the planner is constructed.
[ INFO] [1663926055.087005970]: arm/arm[LazyPRM]: Starting planning with 6675 states already in datastructure
[ INFO] [1663926055.121773308]: arm/arm[LazyPRM]: Created 4294967284 states
[ INFO] [1663926055.121816330]: Solution found in 0.055522 seconds
[ INFO] [1663926055.181362586]: SimpleSetup: Path simplification took 0.049026 seconds and changed from 11 to 2 states
```

--- a/src/ompl/geometric/planners/prm/src/LazyPRM.cpp
+++ b/src/ompl/geometric/planners/prm/src/LazyPRM.cpp
@@ -393,7 +393,7 @@ ompl::base::PlannerStatus ompl::geometric::LazyPRM::solve(const base::PlannerTer
             do
             {
                 solution = constructSolution(startV, goalV);
-            } while (!solution && vertexComponentProperty_[startV] == vertexComponentProperty_[goalV]);
+            } while (!solution && vertexComponentProperty_[startV] == vertexComponentProperty_[goalV] && !ptc);
             if (solution)
             {
                 someSolutionFound = true;

